compiler/rustc_lint/src/noop_method_call.rs

```diff
@@ -114,7 +114,12 @@ impl<'tcx> LateLintPass<'tcx> for NoopMethodCall {
 
         let orig_ty = expr_ty.peel_refs();
 
-        if receiver_ty == expr_ty {
+        if receiver_ty == expr_ty
+            && matches!(
+                name,
+                sym::noop_method_borrow | sym::noop_method_clone | sym::noop_method_deref
+            )
```

Comment:
I think I understand the need for this check now. Could you move this up so that we return early if the name doesn't match any of the noop methods that we want?